lang_php/analyze/php_to_generic.ml

```diff
@@ -263,11 +265,16 @@ and expr =
   | Obj_get (v1, t, Id [v2]) ->
       let v1 = expr v1 and v2 = ident v2 in
       G.DotAccess (v1, t, G.EId (v2, G.empty_id_info()))
+  | Obj_get (v1, t, Metavar v2) ->
+      let v1 = expr v1 and v2 = ident v2 in
+      G.DotAccess (v1, t, G.EId (v2, G.empty_id_info()))
   | Obj_get (v1, t, v2) ->
       let v1 = expr v1 and v2 = expr v2 in
       G.DotAccess (v1, t, G.EDynamic v2)
   | Class_get (v1, t, Id [v2]) -> let v1 = expr v1 and v2 = ident v2 in
       G.DotAccess (v1, t, G.EId (v2, G.empty_id_info()))
+  | Class_get (v1, t, Metavar v2) -> let v1 = expr v1 and v2 = ident v2 in
```

Comment:
yep, same here, there is a complex pattern where Id is matched deep, and we want to also handle
the Var when is_metavar_name && sgrep_mode.

Maybe there are more places in php_to_generic.ml where we match Id deeply; Check for it.

Reply:
I've seen one for the Global case for example (and UseOuterDecl)
maybe we also need to handle specially Var (or your Metavar) there.

Reply:
There's also one in the code to transform attribute. Basically, each time we match Id, we need
to add a case for Var when the var is a metavar.